e2e2/test/cases/netpol/main_test.go

```diff
@@ -0,0 +1,333 @@
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+var testenv env.Environment
+
+func TestMain(m *testing.M) {
+	cfg, err := envconf.NewFromFlags()
+	if err != nil {
+		log.Fatalf("failed to initialize test environment: %v", err)
+	}
+	testenv = env.NewWithConfig(cfg)
+
+	testenv.Setup(
+
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+			if err != nil {
+				return ctx, err
+			}
+
+			namespaces := []string{"a", "b", "c"}
+
+			servers := map[string]string{
+				"a": "a-server",
+				"b": "b-server",
+				"c": "c-server",
+			}
+
+			// 1. Install Latest CNI version
+			err = installLatestCNI(config)
+			if err != nil {
+				log.Print("Failed to install latest aws-vpc-cni on cluster")
+				return ctx, err
+			}
+
+			// 2. Create three namespaces
+			for _, ns := range namespaces {
+				err = createNamespace(ns, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create namespace %s", ns)
+					return ctx, err
+				}
+			}
+
+			// 3. Create deployment and service
+			for ns, server := range servers {
+				err = createServerAndService(ns, server, 1, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create deployment and service for %s", server)
+					return ctx, err
+				}
+			}
+
+			return ctx, nil
+		},
+	)
+
+	testenv.Finish(
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+
+			if err != nil {
+				return ctx, err
+			}
+
+			client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "a", Namespace: "a"}})
+			client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "b", Namespace: "b"}})
+			client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "c", Namespace: "c"}})
+
+			return ctx, nil
+		},
+	)
+
+	os.Exit(testenv.Run(m))
+}
+
+func installLatestCNI(config *envconf.Config) error {
+
+	repoName := "eks"
+	repoUrl := "https://aws.github.io/eks-charts"
+	chartName := "eks/aws-vpc-cni"
+	releaseName := "aws-vpc-cni"
+	releaseNamespace := "kube-system"
+
+	resources := []string{"daemonset/aws-node", "clusterRole/aws-node", "clusterRoleBinding/aws-node", "serviceAccount/aws-node", "configMap/amazon-vpc-cni"}
+
+	for _, resource := range resources {
+		if _, err := fwext.ResourceExists(resource, releaseNamespace, config.KubeconfigFile()); err == nil {
+			log.Printf("Patching resource: %s labels and annotations..", resource)
+			fwext.Label(resource, "app.kubernetes.io/managed-by=Helm", releaseNamespace, config.KubeconfigFile())
+			fwext.Annotate(resource, fmt.Sprintf("meta.helm.sh/release-name=%s", releaseName), releaseNamespace, config.KubeconfigFile())
+			fwext.Annotate(resource, fmt.Sprintf("meta.helm.sh/release-namespace=%s", releaseNamespace), releaseNamespace, config.KubeconfigFile())
+		} else {
+			log.Printf("Resource %s does not exist, skipping patching labels/annotations..", resource)
+		}
+	}
```

Comment:
What's the goal of this?

Reply:
So vpc-cni is installed by default on a cluster and we need to import the resource into the helm chart (apply annotations/labels).  More details here -
https://github.com/aws/eks-charts/tree/master/stable/aws-vpc-cni#adopting-the-existing-aws-node-resources-in-an-eks-cluster

I've updated this to delete the resources instead of patching for labels/annotations